### PR TITLE
Update domain class environment-model

### PR DESCRIPTION
**Pull request by jtdemer**
## Type of change

main work was done by Mirco:

- Adjustments regarding name convention
- add sub categories
- add string enums
- add Georefernce shape

I ajust the the foldername regarding the lived naming convention

## How Has This Been Tested?

- Shacl File was uploaded into the self creation wizard

## Checklist

- [x] My code follows the modelling guidelines of this project
- [x] I have performed a self-review of my own changes

### CODE REVIEW

File: environmentmodel/environment-model_ontology.ttl [hunk 1]
```diff
@@ -1,17 +1,16 @@
 @prefix dcterms: <http://purl.org/dc/terms/> .
-@prefix environment-model: <https://github.com/GAIA-X4PLC-AAD/ontology-management-base/tree/main/environment-model/> .
+@prefix environmentmodel: <https://github.com/GAIA-X4PLC-AAD/ontology-management-base/tree/main/environmentmodel/> .
```
Comment (robertschubert) [inline environmentmodel/environment-model_ontology.ttl:2]: it is allowed to have a '-' in the prefix. I would prefer `environment-model` :)
Already documented in the Readme in the modelling guidelines.

File: environmentmodel/environment-model_ontology.ttl [hunk 2]
```diff
@@ -1,17 +1,16 @@
...
-    rdfs:label "ontology definition for environment-model"@en ;
+environmentmodel: a owl:Ontology ;
+    rdfs:label "ontology definition for environmentModel"@en ;
     dcterms:contributor "Mirco Nierenz (TG)" ;
     owl:versionInfo "0.1"^^xsd:float .
 
-environment-model:Asset a owl:Class ;
-    rdfs:label "environment-model" ;
-    rdfs:comment "attributes for 3d environment models"@en ;
+environmentmodel:environmentModel a owl:Class ;
```
Comment (robertschubert) [inline environmentmodel/environment-model_ontology.ttl:13]: Classnames PascalCase, attributes camelCase
-->EnvironmentModel would be correct.

File: environmentmodel/environment-model_shacl.ttl [hunk 3]
```diff
@@ -0,0 +1,288 @@
+@prefix environmentmodel: <https://github.com/GAIA-X4PLC-AAD/ontology-management-base/tree/main/environmentmodel/> .
+@prefix sh: <http://www.w3.org/ns/shacl#> .
+@prefix skos: <http://www.w3.org/2004/02/skos/core#> .
+@prefix xsd: <http://www.w3.org/2001/XMLSchema#> .
+
+environmentmodel:environmentModelShape a sh:NodeShape ;
```
Comment (robertschubert) [inline environmentmodel/environment-model_shacl.ttl:6]: Shape in PascalCase ->EnvironmentModelShape

File: environmentmodel/environment-model_shacl.ttl [hunk 4]
```diff
@@ -0,0 +1,288 @@
...
+    sh:property [ sh:maxCount 1 ;
+            sh:minCount 1 ;
+            sh:node environmentmodel:QuantityShape ;
+            sh:order 2 ;
+            sh:path environmentmodel:Quantity ],
```
Comment (robertschubert) [inline environmentmodel/environment-model_shacl.ttl:11]: attributes should be camelCase
-> `sh:path environmentmodel:quantity`

File: environmentmodel/environment-model_shacl.ttl [hunk 5]
```diff
@@ -0,0 +1,288 @@
...
+        [ sh:maxCount 1 ;
+            sh:minCount 1 ;
+            sh:node environmentmodel:QualityShape ;
+            sh:order 3 ;
+            sh:path environmentmodel:Quality ],
```
Comment (robertschubert) [inline environmentmodel/environment-model_shacl.ttl:16]: attributes should be camelCase 
--> sh:path environmentmodel:quality

Please correct this for the others also.

File: environmentmodel/environment-model_shacl.ttl [hunk 6]
```diff
@@ -0,0 +1,288 @@
...
+            sh:minCount 1 ;
+            sh:node environmentmodel:FormatShape ;
+            sh:order 1 ;
+            sh:path environmentmodel:Format ],
+        [ sh:maxCount 1 ;
+            sh:minCount 1 ;
+            sh:node environmentmodel:GeoreferenceShape ;
+            sh:order 7 ;
+            sh:path environmentmodel:Georeference ] ;
+    sh:targetClass environmentmodel:environmentModel .
```
Comment (robertschubert) [inline environmentmodel/environment-model_shacl.ttl:32]: PascalCase class
-> `EnvironmentModel`

Please correct this for the others also

File: environmentmodel/environment-model_shacl.ttl [hunk 7]
```diff
@@ -0,0 +1,288 @@
...
+            sh:minCount 1 ;
+            sh:node environmentmodel:ProjectLocationShape ;
+            sh:order 0 ;
+            sh:path environmentmodel:ProjectLocation ],
+        [ sh:maxCount 1 ;
+            sh:minCount 1 ;
+            sh:node environmentmodel:GeodeticReferenceSystemShape ;
+            sh:order 1 ;
+            sh:path environmentmodel:GeodeticReferenceSystem ] ;
+    sh:targetClass environmentmodel:georeference .
```
Comment (robertschubert) [inline environmentmodel/environment-model_shacl.ttl:45]: georeference -> GeoReference

File: environmentmodel/environment-model_shacl.ttl [hunk 8]
```diff
@@ -0,0 +1,288 @@
...
+        [ sh:datatype xsd:float ;
+            sh:message "Validation of y failed!"@en ;
+            sh:name "y"@en ;
+            sh:maxCount 1 ;
+            sh:order 1 ;
+            sh:path environmentmodel:y ] ;
+    sh:targetClass environmentmodel:Coordinate2D .
+    
+environmentmodel:ProjectLocationShape a sh:NodeShape ;
+    sh:property [ skos:example "nan" ;
```
Comment (robertschubert) [inline environmentmodel/environment-model_shacl.ttl:118]: if there is no valid example I would remove the example with "nan"

Please correct this for the others too

File: environmentmodel/environment-model_shacl.ttl [hunk 9]
```diff
@@ -0,0 +1,288 @@
...
+            sh:path environmentmodel:city ],
+        [ skos:example "nan" ;
+            sh:datatype xsd:string ;
+            sh:description "Region of project area"@en ;
+            sh:message "Validation of region failed!"@en ;
+            sh:maxCount 1 ;
+            sh:name "region"@en ;
+            sh:order 2 ;
+            sh:path environmentmodel:region ],
+        [ skos:example "Germany, GER" ;
```
Comment (robertschubert) [inline environmentmodel/environment-model_shacl.ttl:142]: cant you use the address shape which is already in the gaia-x vocabulary?

<http://www.w3.org/2006/vcard/ns#Address>

if yes you can add it as a shape with following approach:
```
[ sh:minCount 1 ;
            sh:node <http://www.w3.org/2006/vcard/ns#Address> ;
            sh:path environmentmodel:address;
            sh:order 5 ] ;
```
Or you could use PostalAddressShape:
```
[ sh:minCount 1 ;
            sh:node gx:PostalAddressShape ;
            sh:path environmentmodel:address;
            sh:order 5 ] ;
```
But this has only one property for a  ISO 3166-2 based string.

File: environmentmodel/environment-model_ontology.ttl [hunk 10]
```diff
@@ -1,17 +1,16 @@
 @prefix dcterms: <http://purl.org/dc/terms/> .
-@prefix environment-model: <https://github.com/GAIA-X4PLC-AAD/ontology-management-base/tree/main/environment-model/> .
+@prefix environmentmodel: <https://github.com/GAIA-X4PLC-AAD/ontology-management-base/tree/main/environmentmodel/> .
 @prefix gax-core: <https://w3id.org/gaia-x/core#> .
 @prefix owl: <http://www.w3.org/2002/07/owl#> .
 @prefix rdfs: <http://www.w3.org/2000/01/rdf-schema#> .
 @prefix xsd: <http://www.w3.org/2001/XMLSchema#> .
```
Comment (robertschubert) [inline environmentmodel/environment-model_ontology.ttl:6]: Please add 
@prefix gx: <https://registry.lab.gaia-x.eu/development/api/trusted-shape-registry/v1/shapes/jsonld/trustframework#> .

File: environmentmodel/environment-model_ontology.ttl [hunk 11]
```diff
@@ -1,17 +1,16 @@
...
     dcterms:contributor "Mirco Nierenz (TG)" ;
     owl:versionInfo "0.1"^^xsd:float .
 
-environment-model:Asset a owl:Class ;
-    rdfs:label "environment-model" ;
-    rdfs:comment "attributes for 3d environment models"@en ;
+environmentmodel:environmentModel a owl:Class ;
+    rdfs:label "class definition for environmentModel" ;
+    rdfs:comment "Attributes for 3d environment models"@en ;
     rdfs:subClassOf gax-core:Resource .
```
Comment (robertschubert) [inline environmentmodel/environment-model_ontology.ttl:16]: should be gx:DataResource

File: environmentmodel/environment-model_ontology.ttl (hunk 1 above)
Comment (jtdemer) [reply environmentmodel/environment-model_ontology.ttl]: I totally agree on that:) 
It should be changed now.

File: environmentmodel/environment-model_ontology.ttl (hunk 2 above)
Comment (jtdemer) [reply environmentmodel/environment-model_ontology.ttl]: I adapted that.

File: environmentmodel/environment-model_shacl.ttl (hunk 3 above)
Comment (jtdemer) [reply environmentmodel/environment-model_shacl.ttl]: I adapted that.

File: environmentmodel/environment-model_shacl.ttl (hunk 4 above)
Comment (jtdemer) [reply environmentmodel/environment-model_shacl.ttl]: I adapted that.

File: environmentmodel/environment-model_shacl.ttl (hunk 5 above)
Comment (jtdemer) [reply environmentmodel/environment-model_shacl.ttl]: I adapt hopefully all.

File: environmentmodel/environment-model_shacl.ttl (hunk 7 above)
Comment (jtdemer) [reply environmentmodel/environment-model_shacl.ttl]: I adapted that.

File: environmentmodel/environment-model_shacl.ttl (hunk 8 above)
Comment (jtdemer) [reply environmentmodel/environment-model_shacl.ttl]: @MircoNierenz can you look into that topic? I think currently it is a placeholder.
@robertschubert can merge this branch allready into main with this placeholder?

File: environmentmodel/environment-model_shacl.ttl (hunk 9 above)
Comment (jtdemer) [reply environmentmodel/environment-model_shacl.ttl]: As far as I know, the idea here was to use the region output from the geodatabase. However, we can take another look at this in the data provider group.

@MircoNierenz @3Dbastian

File: environmentmodel/environment-model_ontology.ttl (hunk 10 above)
Comment (jtdemer) [reply environmentmodel/environment-model_ontology.ttl]: I adapted that.

File: environmentmodel/environment-model_shacl.ttl (hunk 8 above)
Comment (robertschubert) [reply environmentmodel/environment-model_shacl.ttl]: I think we can merge without having a solution right now.
Opened Issue https://github.com/GAIA-X4PLC-AAD/ontology-management-base/issues/36

File: environmentmodel/environment-model_shacl.ttl (hunk 9 above)
Comment (robertschubert) [reply environmentmodel/environment-model_shacl.ttl]: Opened a new issue:
https://github.com/GAIA-X4PLC-AAD/ontology-management-base/issues/37

**Review by robertschubert — APPROVED** [review]

**Comment by robertschubert** [comment]
@jtdemer 
Looks good :)
One thing I did not see in the first review: 
We need the functional responsible person for this ontology. Can you figure out who this can be? Can you add a section into the CODEOWNERs file? 
After this an a second approval by this person this PR can be merged.

**Comment by robertschubert** [comment]
@jtdemer can you also rename the example file?
it should be written as ontology and shacl followed with an underscore ("_instance.json").
environment-model_instance.json.

File: environment-model/environment-model_shacl.ttl [hunk 12]
```diff
@@ -3,134 +3,286 @@
...
+            sh:name "y"@en ;
+            sh:maxCount 1 ;
+            sh:order 1 ;
+            sh:path environment-model:y ] ;
+    sh:targetClass environment-model:Coordinate2D .
+    
+environment-model:ProjectLocationShape a sh:NodeShape ;
+    sh:property [ skos:example "nan" ;
+            sh:datatype xsd:string ;
+            sh:description "State of project area"@en ;
```
Comment (jdsika) [inline environment-model/environment-model_shacl.ttl:120]: @jtdemer we have ISO standards for this!

https://en.wikipedia.org/wiki/ISO_3166-2


Please rework all values according to proper ISO definitions if avaliable

File: environment-model/environment-model_shacl.ttl [hunk 13]
```diff
@@ -3,134 +3,286 @@
...
+    sh:targetClass environment-model:ProjectLocation .     
+
+environment-model:FormatShape a sh:NodeShape ;
+    sh:property [ skos:example "Unreal DataSmith, Autodesk FBX, OpenSceneGraph" ;
+            sh:datatype xsd:string ;
+            sh:description "Format type definition"@en ;
+            sh:in ("Autodesk FBX" "GLTF" "OpenSceneGraph" "Unreal DataSmith") ;
+            sh:maxCount 1 ;
+            sh:message "Validation of type failed!"@en ;
+            sh:name "type"@en ;
```
Comment (jdsika) [inline environment-model/environment-model_shacl.ttl:177]: Is "type" the right name here?

Comment (jdsika) [reply environment-model/environment-model_shacl.ttl:177]: below is "verison of data format". I think it refers to e.g. GLTF? - in this case "dataFormat" would make more sense then "type"?

File: environment-model/environment-model_shacl.ttl [hunk 14]
```diff
@@ -3,134 +3,286 @@
...
+            sh:message "Validation of creationSource failed!"@en ;
+            sh:name "creationSource"@en ;
+            sh:order 3 ;
+            sh:path environment-model:creationSource ],
+        [ skos:example "Unreal" ;
+            sh:datatype xsd:string ;
+            sh:description "Name of the visual system"@en ;
+            sh:maxCount 1 ;
+            sh:message "Validation of softwareName failed!"@en ;
+            sh:name "softwareName"@en ;
```
Comment (jdsika) [inline environment-model/environment-model_shacl.ttl:212]: graphicsEngine? What is the right name for this? At least wikipedia calls it an engine https://de.wikipedia.org/wiki/Unreal_Engine

File: environment-model/environment-model_shacl.ttl [hunk 15]
```diff
@@ -3,134 +3,286 @@
...
+            sh:order 0 ;
+            sh:path environment-model:softwareName ],
         [ skos:example "Epic" ;
```
Comment (jdsika) [inline environment-model/environment-model_shacl.ttl:215]: It is called "Epic Games". We really need to be more precise with a lot of things here

File: environment-model/environment-model_shacl.ttl [hunk 16]
```diff
@@ -3,134 +3,286 @@
...
-            sh:path environment-model:environment-model_quantity_tile_count_y ],
-        [ skos:example "179" ;
-            sh:datatype xsd:unsignedInt ;
-            sh:description "number of textures"@en ;
-            sh:maxCount "1"^^xsd:unsignedInt ;
-            sh:message "Validation of texture_material_count failed!"@en ;
-            sh:minCount "1"^^xsd:unsignedInt ;
-            sh:name "texture_material_count"@en ;
-            sh:path environment-model:environment-model_quantity_texture_material_count ],
+            sh:description "Total number of all geoemtries, instances are considered only once"@en ;
```
Comment (jdsika) [inline environment-model/environment-model_shacl.ttl:265]: Is there somewhere a definition of this? "geometries according to "

**Comment by MircoNierenz** [comment]
my new version was not commited, was it? 
I have now done this and also the adjustments from the other branches. Please have a look.

**Comment by robertschubert** [comment]
> my new version was not commited, was it? I have now done this and also the adjustments from the other branches. Please have a look.

@MircoNierenz 
there is no commit from you (see commit tab on the top):

image

@jtdemer 
we still need a function owner in the CODEOWNERS file (see comment from May 22nd.

**Comment by jtdemer** [comment]
> > my new version was not commited, was it? I have now done this and also the adjustments from the other branches. Please have a look.
> 
> @MircoNierenz there is no commit from you (see commit tab on the top):
> 
> image
> Please give me a short information if you need help.
> 
> @jtdemer we still need a function owner in the CODEOWNERS file (see comment from May 22nd.

done

**Comment by MircoNierenz** [comment]
> > my new version was not commited, was it? I have now done this and also the adjustments from the other branches. Please have a look.
> 
> @MircoNierenz there is no commit from you (see commit tab on the top):

I have commited to the new environment-model branch!

File: environment-model/environment-model_shacl.ttl (hunk 12 above)
Comment (MircoNierenz) [reply environment-model/environment-model_shacl.ttl:120]: outdated. 
moved to Georeference shacl . see georeference:country

File: environment-model/environment-model_shacl.ttl (hunk 13 above)
Comment (MircoNierenz) [reply environment-model/environment-model_shacl.ttl:177]: change to dataType in category format

File: environment-model/environment-model_shacl.ttl (hunk 14 above)
Comment (MircoNierenz) [reply environment-model/environment-model_shacl.ttl:212]: change to graphics engine

File: environment-model/environment-model_shacl.ttl (hunk 15 above)
Comment (MircoNierenz) [reply environment-model/environment-model_shacl.ttl:215]: fixed

**Review by MircoNierenz — COMMENTED** [review]
ok for merge

File: environment-model/environment-model_shacl.ttl [hunk 17]
```diff
@@ -3,134 +3,286 @@
...
+        [ sh:datatype xsd:float ;
+            sh:message "Validation of y failed!"@en ;
+            sh:name "y"@en ;
+            sh:maxCount 1 ;
+            sh:order 1 ;
+            sh:path environment-model:y ] ;
+    sh:targetClass environment-model:Coordinate2D .
+    
+environment-model:ProjectLocationShape a sh:NodeShape ;
+    sh:property [ skos:example "nan" ;
```
Comment (robertschubert) [inline environment-model/environment-model_shacl.ttl:118]: still some "nan" examples here.
Please provide an example or delete the example

File: environment-model/environment-model_shacl.ttl [hunk 18]
```diff
@@ -3,134 +3,286 @@
...
+            sh:order 1 ;
+            sh:path environment-model:format ],
+        [ sh:maxCount 1 ;
+            sh:minCount 1 ;
+            sh:node environment-model:GeoreferenceShape ;
+            sh:order 7 ;
+            sh:path environment-model:georeference ] ;
+    sh:targetClass environment-model:EnvironmentModel .
+
+    environment-model:GeoreferenceShape a sh:NodeShape ;
```
Comment (robertschubert) [inline environment-model/environment-model_shacl.ttl:34]: Is this another GeoreferenceShape then the one which is developed here?
I guess it is the same. We should link it here and not duplicate it.

The _instance.json file then also needs an update.

File: environment-model/environment-model_ontology.ttl [hunk 19]
```diff
@@ -1,17 +1,16 @@
...
+@prefix gx: <https://registry.lab.gaia-x.eu/development/api/trusted-shape-registry/v1/shapes/jsonld/trustframework#> .
 @prefix owl: <http://www.w3.org/2002/07/owl#> .
 @prefix rdfs: <http://www.w3.org/2000/01/rdf-schema#> .
 @prefix xsd: <http://www.w3.org/2001/XMLSchema#> .
 
 environment-model: a owl:Ontology ;
-    rdfs:label "ontology definition for environment-model"@en ;
+    rdfs:label "ontology definition for EnvironmentModel"@en ;
     dcterms:contributor "Mirco Nierenz (TG)" ;
     owl:versionInfo "0.1"^^xsd:float .
```
Comment (robertschubert) [inline environment-model/environment-model_ontology.ttl:11]: ^^xsd:float can be removed

**Review by robertschubert — CHANGES_REQUESTED** [review]
@MircoNierenz 
found 3 issues, please see comments.

File: environment-model/environment-model_shacl.ttl (hunk 17 above)
Comment (MircoNierenz) [reply environment-model/environment-model_shacl.ttl:118]: outdated? see branch environment-model and 
https://github.com/GAIA-X4PLC-AAD/ontology-management-base/pull/44

File: environment-model/environment-model_shacl.ttl (hunk 18 above)
Comment (MircoNierenz) [reply environment-model/environment-model_shacl.ttl:34]: outdated? see branch environment-model and 
https://github.com/GAIA-X4PLC-AAD/ontology-management-base/pull/44

File: environment-model/environment-model_ontology.ttl (hunk 19 above)
Comment (MircoNierenz) [reply environment-model/environment-model_ontology.ttl:11]: outdated? see branch environment-model and 
https://github.com/GAIA-X4PLC-AAD/ontology-management-base/pull/44

**Comment by MircoNierenz** [comment]
please sove pull request 
outdated? see branch environment-model and 
https://github.com/GAIA-X4PLC-AAD/ontology-management-base/pull/44
and merge into main
the environment-model ontology in the main branch is not up to date!

**Comment by robertschubert** [comment]
@MircoNierenz Or is one of the two branches / PRs for environment model obsolete?
Still we need a second approval so I will not do the merge now.

**Comment by MircoNierenz** [comment]
had missed that this pull request is already for branch environment-model - due to the different naming of the PR. I had therefore made my own PR.

then the PR https://github.com/GAIA-X4PLC-AAD/ontology-management-base/pull/44 should be obsolete.

**Comment by robertschubert** [comment]
@MircoNierenz 
Then please close PR https://github.com/GAIA-X4PLC-AAD/ontology-management-base/pull/44 and incorporate the review comments from above into this PR. They are still open and I cannot merge this.

@lenasauermann after the review comments are incorporated can you please do another approval?

**Review by MircoNierenz — APPROVED** [review]
ok for me

**Review by MircoNierenz — APPROVED** [review]
ok for me now

**Review by lenasauermann — APPROVED** [review]

**Comment by robertschubert** [comment]
@MircoNierenz 
I cannot see any commit and the issues are still not resolved in the code. I reopened the conversation.
To summarize:
- still "nan" as example
- "GeoreferenceShape" is defined locally. Shouldn't you take the globally defined? If yes you will also need to update the _instance file
- still `0.1"^^xsd:float` in the ontology file.

File: environment-model/environment-model_ontology.ttl (hunk 19 above)
Comment (robertschubert) [reply environment-model/environment-model_ontology.ttl:11]: still the same in the code.

File: environment-model/environment-model_shacl.ttl (hunk 18 above)
Comment (robertschubert) [reply environment-model/environment-model_shacl.ttl:34]: unresolved, still the same in the code.

File: environment-model/environment-model_shacl.ttl (hunk 17 above)
Comment (robertschubert) [reply environment-model/environment-model_shacl.ttl:118]: unresolved, still the same in the code.

**Comment by robertschubert** [comment]
I can do the "resolve conflicts" for you

**Comment by MircoNierenz** [comment]
this pull request is outdated! reference to wrong branch (environment-model vs Ontology_Environment)!
see new pull request https://github.com/GAIA-X4PLC-AAD/ontology-management-base/pull/50 for Ontology_Environment branch

**Comment by MircoNierenz** [comment]
this pull request is outdated! reference to wrong branch (environment-model vs Ontology_Environment)!
see new pull request https://github.com/GAIA-X4PLC-AAD/ontology-management-base/pull/50 for Ontology_Environment branch

**Pull request closed without merging**